cmd/libsnap-confine-private/snap.c

```diff
@@ -169,25 +169,49 @@ void sc_snap_name_validate(const char *snap_name, struct sc_error **errorp)
 	sc_error_forward(errorp, err);
 }
 
-void sc_snap_drop_instance_name(const char *snap_name, char *base,
-				size_t base_size)
+void sc_snap_drop_instance_name(const char *name, char *snap_name,
+				size_t snap_name_size)
 {
-	if (snap_name == NULL || base == NULL) {
-		die("base or snap name unset");
+	sc_snap_split_instance_name(name, snap_name, snap_name_size, NULL, 0);
+}
+
+void sc_snap_split_instance_name(const char *name, char *snap_name,
+				 size_t snap_name_size, char *instance_key,
+				 size_t instance_key_size)
+{
+	if (name == NULL) {
+		die("name not provided");
+	}
+	if (snap_name == NULL && instance_key == NULL) {
+		die("snap name and instance key are unset");
```

Comment:
This should say something like:

"internal error: cannot split instance name when both snap name and instance key are unset"

and in the earlier case:

"internal error: cannot split snap name when snap name is unset"

Let's please keep an eye on having error messages that the user can relate to (and developers too, in fact).